src/d3a/models/market/market_structures.py

```diff
@@ -54,6 +54,8 @@ class Offer:
     def __post_init__(self):
         self.id = str(self.id)
         self.energy_rate = self.price / self.energy
+        if self.original_offer_price is None:
```

Comment:
For future PRs: I think it would be good to rename `self.original_offer_price` and `self.original_bid_price` to `self.original_price`. This has some advantages:
1. the name would be a bit better, since you remove a repetition coming from the class name (`offer.original_price` vs `offer.original_offer_price`)
2. it would allow homogenizing `Offer` and `Bid` objects' behaviour (and maybe someday they'll share the same interface)

Reply:
I totally agree, I also suggested this in a slack conversation , I may do it in a next PR